### PR TITLE
chore: ask prompts for `local run`

### PR DESCRIPTION
**Pull request by Varun359**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/deploy/deploy.go [hunk 1]
```diff
@@ -312,6 +317,11 @@ func (s *Store) IsJobDeployed(appName, envName, jobName string) (bool, error) {
 	return s.isWorkloadDeployed(appName, envName, jobName)
 }
 
+// IsWorkloadDeployed returns whether a workload is deployed in an environment or not.
+func (s *Store) IsWorkloadDeployed(appName, envName, wkldName string) (bool, error) {
+	return s.isWorkloadDeployed(appName, envName, wkldName)
+}
+
```
Comment (iamhopaul123) [inline internal/pkg/deploy/deploy.go]: Why not just expose `isWorkloadDeployed`?

File: internal/pkg/term/selector/selector.go [hunk 2]
```diff
@@ -609,7 +627,9 @@ func (s *DeploySelector) deployedWorkload(workloadType string, msg, help string,
 		listDeployedWorkloads = s.deployStoreSvc.ListDeployedJobs
 		finalMessage = deployedJobFinalMsg
 	default:
```
Comment (iamhopaul123) [inline internal/pkg/term/selector/selector.go]: I would suggest make this more explicit: right now we get to default when `workloadType` is "workload". Maybe we can still error out for default and add a new case for `workloadType`?

File: internal/pkg/term/selector/selector.go [hunk 3]
```diff
@@ -23,6 +23,7 @@ import (
 const (
 	svcWorkloadType = "service"
 	jobWorkloadType = "job"
+	workloadType    = "workload"
```
Comment (iamhopaul123) [inline internal/pkg/term/selector/selector.go]: Maybe `wkldWorkloadType` to be more consistent 🤔 Also avoid conflating with the parameter name for `workloadType`.

**Review by iamhopaul123 — COMMENTED** [review]
Awesome!

**Review by iamhopaul123 — APPROVED** [review]
🚢

File: internal/pkg/cli/local_run.go [hunk 4]
```diff
@@ -56,8 +67,32 @@ func (o *localRunOpts) Validate() error {
 
 // Ask prompts the user for any unprovided required fields and validates them.
 func (o *localRunOpts) Ask() error {
-	//TODO(varun359): Validate and Ask SvcEnvName
+	if err := o.validateAndAskWkldEnvName(); err != nil {
+		return err
+	}
+
+	return nil
```
Comment (Lou1415926) [inline internal/pkg/cli/local_run.go]: If we don't have more prompts to add:
```suggestion
	return  o.validateAndAskWkldEnvName()
```

File: internal/pkg/cli/local_run.go [hunk 5]
```diff
@@ -56,8 +67,32 @@ func (o *localRunOpts) Validate() error {
...
+	}
+	if o.wkldName != "" {
+		if _, err := o.store.GetWorkload(o.appName, o.wkldName); err != nil {
+			return err
+		}
+	}
+
+	deployedWorkload, err := o.sel.DeployedWorkload(workloadAskPrompt, "", o.appName, selector.WithEnv(o.envName), selector.WithName(o.wkldName))
+	if err != nil {
+		return fmt.Errorf("select deployed workloads for application %s: %w", o.appName, err)
```
Comment (Lou1415926) [inline internal/pkg/cli/local_run.go]: nit:
```suggestion
		return fmt.Errorf("select a deployed workload from application %s: %w", o.appName, err)
```

File: internal/pkg/cli/local_run_test.go [hunk 6]
```diff
@@ -74,3 +76,109 @@ func TestLocalRunOpts_Validate(t *testing.T) {
...
+		testEnvName  = "testEnv"
+		testWkldName = "testWkld"
+	)
+	testCases := map[string]struct {
+		inputAppName  string
+		inputEnvName  string
+		inputWkldName string
+
+		setupMocks     func(m *localRunAskMocks)
+		mockEnvChecker func(ctrl *gomock.Controller) versionCompatibilityChecker
```
Comment (Lou1415926) [inline internal/pkg/cli/local_run_test.go]: Is this needed 👀 ?

File: internal/pkg/cli/local_run_test.go [hunk 7]
```diff
@@ -74,3 +76,109 @@ func TestLocalRunOpts_Validate(t *testing.T) {
...
+		wantedWkldName string
+		wantedEnvName  string
```
Comment (Lou1415926) [inline internal/pkg/cli/local_run_test.go:94]: noice!! Can we also have a `wantedWkldType string` 🥺

File: internal/pkg/term/selector/selector.go [hunk 8]
```diff
@@ -21,8 +21,9 @@ import (
 )
 
 const (
-	svcWorkloadType = "service"
-	jobWorkloadType = "job"
+	svcWorkloadType  = "service"
+	jobWorkloadType  = "job"
+	wkldWorkloadType = "workload"
```
Comment (Lou1415926) [inline internal/pkg/term/selector/selector.go]: nit
```suggestion
	anyWorkloadType = "workload"
```

**Review by Lou1415926 — APPROVED** [review]
very noice! approved with the label - feel free to just remove it once the feedbacks are addressed!

**Pull request merged**